### DISCUSSION TITLE
Problem using Sikuli with robotMK

### DISCUSSION THREAD

**Discussion by jaketehpwner** [elabit/robotmk#202] [Q&A] [answered]
Hello,

I am trying to automate an application launched from a browser in robotMK. The selenium stuff seems to be working but the image recognition piece with Sikuli is failing because it is only seeing black images when it takes a screenshot. When I run the test outside of the robotMK runner it works great. Has anyone seen this or have a workaround?

Kind regards,
Jake

**Comment by simonmeggle** [comment, marked as answer]
Hi Jake, sounds to me like a problem with the desktop. Sikuli (as well as the ImagehorizonLibrary, which I prefer) need acess to the (physical) desktop. When Robotmk launches the test, I guess it runs with a different user than the one which is logged on the desktop. 
When you are launching the test by hand, you _are_ on the desktop - and this is why sikuli "sees" something. 

My recommendations: 
- switch to [imagehorizonlibrary](https://github.com/eficode/robotframework-imagehorizonlibrary/) (and, if possible, leave a thumbs up to this [pull request](https://github.com/eficode/robotframework-imagehorizonlibrary/pull/57) :-) )
- use Robotmk mode "external" and schedule `robotmk-runner.py` by Task Scheduler from the user which is logged on.

**Comment by jaketehpwner** [comment]
Thanks Simon,

Running from task scheduler seems to do the trick. I will also check out imagehorizonlibrary.
